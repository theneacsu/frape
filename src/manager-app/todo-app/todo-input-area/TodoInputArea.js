import React from 'react'
import { connect } from 'react-redux'
import TodoForm from '../todo-form/TodoForm'
import { startAddTodo } from '../../../actions/todo-app/todos'

export const TodoInputArea = props => {
  const onSubmitAddTodo = todo => props.startAddTodo(todo.text)

  return (
    <div className="todo-input-area-box">
      <TodoForm
        placeholder="ex: Go for a run"
        buttonText="Add todo"
        onSubmit={onSubmitAddTodo}
      />
    </div>
  )
}

const mapDispatchToProps = dispatch => ({
  startAddTodo: text => dispatch(startAddTodo(text))
})

export default connect(
  undefined,
  mapDispatchToProps
)(TodoInputArea)
