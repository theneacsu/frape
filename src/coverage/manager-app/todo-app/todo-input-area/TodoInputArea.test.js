import React from 'react'
import { shallow } from 'enzyme'
import { TodoInputArea } from '../../../../manager-app/todo-app/todo-input-area/TodoInputArea'

test('Should render correctly the TodoInputArea component', () => {
  const wrapper = shallow(<TodoInputArea />)
  expect(wrapper).toMatchSnapshot()
})

test('Should call the addTodo action when the form is submited with valid data', () => {
  const addTodoSpy = jest.fn()
  const todo = {
    text: 'Drink a latte',
    completed: false
  }
  const wrapper = shallow(<TodoInputArea addTodo={addTodoSpy} />)
  wrapper.find('TodoForm').prop('onSubmit')(todo)
  expect(addTodoSpy).toHaveBeenLastCalledWith(todo.text)
})
