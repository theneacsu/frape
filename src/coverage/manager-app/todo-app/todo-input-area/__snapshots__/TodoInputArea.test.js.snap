// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly the TodoInputArea component 1`] = `
<div
  className="todo-input-area-box"
>
  <TodoForm
    buttonText="Add todo"
    onSubmit={[Function]}
    placeholder="ex: Go for a run"
  />
</div>
`;
